monetio/sat/_tropomi_l2_no2_mm.py

```diff
@@ -54,45 +60,120 @@ def _open_one_dataset(fname, variable_dict):
     ds["scan_time"] = ds["time"] + dtime
     ds["scan_time"].attrs.update({"long_name": "scan time"})
     ds = ds.set_coords(["lon", "lat", "time", "scan_time"])
-    ds.attrs["reference_time_string"] = ref_time_val.astype(datetime).strftime(r"%Y%m%d")
+    ds.attrs["reference_time_string"] = ref_time_val.astype(datetime).strftime(r"%Y-%m-%d")
+
+    def get_extra(varname_, *, dct_=None, default_group="PRODUCT"):
+        """Get non-varname variables."""
+        if dct_ is None:
+            dct_ = variable_dict.get(varname_, {})
+        group_name = dct_.get("group", default_group)
+        if isinstance(group_name, list):
+            group_name = group_name[0]
+        return _get_values(dso[group_name][varname_], dct_)
+
+    for varname, dct in variable_dict.items():
+        print(f"- {varname}")
+        if dct is None:
+            dct = {}
+
+        if varname == "preslev":
+            # Compute layered pressure and tropopause pressure
+
+            itrop_ = get_extra("tm5_tropopause_layer_index", dct_=dct)
+            itrop_[itrop_ == 0] = 1  # Avoid trop in surface layer
+            itrop = xr.DataArray(itrop_, dims=("y", "x"))
+            a = xr.DataArray(data=get_extra("tm5_constant_a", dct_=dct), dims=("z", "v"))
+            b = xr.DataArray(data=get_extra("tm5_constant_b", dct_=dct), dims=("z", "v"))
+            psfc = xr.DataArray(
+                data=get_extra(
+                    "surface_pressure",
+                    default_group="PRODUCT/SUPPORT_DATA/INPUT_DATA",
+                    dct_=dct,
+                ),
+                dims=("y", "x"),
+            )
+
+            # Mid-layer pressure
+            assert a.sizes["v"] == 2, "base and top"
+            p = ((a.isel(v=0) + b.isel(v=0) * psfc) + (a.isel(v=1) + b.isel(v=1) * psfc)) / 2
```

Comment:
can you change the dimension from (z,y,x,z) to (y,x,z)? This dimension is consistent with the follow-up process of TROPOMI.

p = p.transpose('y','x','z')

Reply:
@mlirenzhenmayi Would this order be desirable for _camx_mm as well?

Reply:
> @mlirenzhenmayi Would this order be desirable for _camx_mm as well?

I'm really not sure. Have not looked into _camx_mm before. If you're using the same averaging kernel correction processing like TROPOMI-WRF_Chem, then I suppose the dimension should be ('y','x','z') to avoid errors

Reply:
I had only worked on it for TEMPO, but if only that and the layer thickness are needed (we already have there the other functions) it seems to me like an easy change that would get it working with the code you wrote for TROPOMI as well.
If you think that should work, I am happy to do the changes in CAMx and get a more consistent codebase

Reply:
I believe that for model data for MM at least, (time, z, y, x) is the "standard".

An alternative approach @mlirenzhenmayi would be to add the `.transpose('y','x','z')` in the start of your function, to ensure dims always match your expectation. Or, with xarray it should be able to write things that don't require a certain dim order.

Reply:
> I believe that for model data for MM at least, (time, z, y, x) is the "standard".
> 
> An alternative approach @mlirenzhenmayi would be to use the `.transpose('y','x','z')` near the start of your processing function, to ensure dims always match your expectation. Or, with xarray it should be able to write things that don't require a certain dim order.

OK. I can revise the processing module to transpose the dimensions. Everything looks good to me now.

Reply:
Ok, thanks a lot!